File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/spring_boot/SpringBootSettings.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.spring_boot;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.github.alexandrecarlton.idea.settings.layout.configurations.ConfigurationSettings;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableSpringBootSettings.class)
public interface SpringBootSettings extends ConfigurationSettings {

  String name();

  SpringBootConfigurationSettings configuration();

}

File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/spring_boot/SpringBootSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.configurations.spring_boot

import com.github.alexandrecarlton.idea.settings.layout.configurations.ConfigurationSettings
import com.github.alexandrecarlton.idea.settings.layout.configurations.common.before_launch.BeforeLaunchConfigurationSettings

data class SpringBootSettings (
    override val name: String,

    override val shareThroughVcs: Boolean? = null,

    override val beforeLaunch: List<BeforeLaunchConfigurationSettings>? = null,

    val configuration: SpringBootConfigurationSettings
) : ConfigurationSettings
